controller/user.go

```diff
@@ -33,20 +39,21 @@ type UserControllerConfiguration interface {
 }
 
 // NewUserController creates a user controller.
-func NewUserController(service *goa.Service, config UserControllerConfiguration) *UserController {
+func NewUserController(service *goa.Service, db application.DB, config UserControllerConfiguration, httpOptions ...configuration.HTTPClientOption) *UserController {
 	return &UserController{
-		Controller: service.NewController("UserController"),
-		config:     config,
+		Controller:  service.NewController("UserController"),
+		config:      config,
+		db:          db,
+		httpOptions: httpOptions,
 	}
 }
 
 // Show returns the authorized user based on the provided Token
 func (c *UserController) Show(ctx *app.ShowUserContext) error {
-	client, err := c.createClient(ctx)
+	client, err := newAuthClient(ctx, c.config, c.httpOptions...)
```

Comment:
Well... it's not required for this PR but this Show method could be cleaned up.. There is no need to use Auth client here. Pure proxy will work just fine (see how it's done in /users controller). Originally we didn't switch to the proxy for /user because we need to call tenant init() here. But it's now done in Auth anyway. So, the whole Show() method could be just:

```
func (c *UsersController) Show(ctx *app.ShowUserContext) error {
	return proxy.RouteHTTP(ctx, c.config.GetAuthShortServiceHostName())
}
```

Reply:
good point, but I'd rather open another PR for that.